### PR TITLE
Split Mixtral test into two scripts

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Split Mixtral end-to-end test into 2 scripts, and this PR works with XL ML [PR](https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/262).

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** ...
Build image, and upload to Airflow and test.

**List links for your tests (use go/shortn-gen for any internal link):** ...
Airflow test: [link](https://screenshot.googleplex.com/7tENx9CsQ3rE6Bk)
XPK test: [link](https://pantheon.corp.google.com/kubernetes/service/us-central2/v4-bodaborg/default/maxtext-stable-mixtral-8x7b-v4-19b385d69/logs?e=13803378&mods=allow_workbench_image_override&project=tpu-prod-env-multipod)

### CODE REVIEW

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# This file, combined with step 2 in the same directory, demonstrates converting a Mixtral checkpoint from Mistral AI.
+# This step is tested nightly on an ordinary CPU VM.
```
Comment (gobbleturk) [inline end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh:4]: Nit: Both steps are run together as a nightly test. The first step converts the Mistral pytorch checkpoint to MaxText (orbax) format using only a CPU VM, whereas the second step uses the checkpoint to run inference and fine-tuning on a TPU.

**Review by gobbleturk — APPROVED** [review]
Looks great! I made some suggestions for changing the comments here, feel free to take what you think is helpful for future code readers. I think there is one incomplete sentence with "both" which should be fixed

File: end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Use the same BASE_OUTPUT_PATH for end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh.
+# Please note that in these two scripts (1_test_mixtral.sh and 2_test_mixtral.sh) BASE_OUTPUT_PATH is assumed to be already a unique path across multiple runs and 
+# the subfolders names aka RUN_NAMEs are static. Please remember to change BASE_OUTPUT_PATH across different runs.
+
+set -ex
+MODEL_VARIATION='8x7b'
+
+if [ -z "${BASE_OUTPUT_PATH}" ]; then
+    # Non-Googlers please remember to point BASE_OUTPUT_PATH to GCS buckets that you own, this script uses internal buckets for testing.
+    # Use the same BASE_OUTPUT_PATH for end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh.
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh:20]: Sure, will update it!

File: end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# This file is both an integration test that runs once a day on a v4-128. 
+# Please make sure you have run end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh before running commands from this file. 
+
+# The flow of this file is as follows:
+# 1. Run decoding based on checkpoint from step 1
+# 2. Run training
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh:8]: Thanks! That's true as we passed in `load_parameters_path`

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh (hunk 1 above)
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh:4]: SG! Will update.

File: end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# This file is both an integration test that runs once a day on a v4-128.
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh:3]: Thanks! Let me update it to make less confusion.

**Review by rwitten — APPROVED** [review]
Please have Anisha provide approval here.

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# This file, combined with step 2 in the same directory, demonstrates:
```
Comment (A9isha) [inline end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh:3]: Perhaps mention that we run this test nightly

https://github.com/google/maxtext/blob/db31dd4b0b686bca4cd7cf940917ec372faa183a/end_to_end/tpu/gemma/7b/1_test_gemma.sh#L3C1-L3C2

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+if [ -z "${BASE_OUTPUT_PATH}" ]; then
+    # Non-Googlers please remember to point BASE_OUTPUT_PATH to GCS buckets that you own, this script uses internal buckets for testing.
+    export BASE_OUTPUT_PATH=gs://runner-maxtext-logs/$(date +%Y-%m-%d-%H-%M)/
+    echo "BASE_OUTPUT_PATH is not set, using BASE_OUTPUT_PATH = ${BASE_OUTPUT_PATH}"
+fi
+
+# Download checkpoint, convert it to MaxText(orbax) format
+pip3 install torch
+gcloud storage cp -r gs://maxtext-external/mixtral-8x7B-v0.1-Instruct /tmp
+JAX_PLATFORMS=cpu python3 MaxText/llama_or_mistral_ckpt.py --base-model-path /tmp/mixtral-8x7B-v0.1-Instruct --model-size mixtral-8x7b --maxtext-model-path ${BASE_OUTPUT_PATH}${MODEL_VARIATION}/decode-ckpt-maxtext/
```
Comment (A9isha) [inline end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh:24]: Do we not want to have an unscanned version of the checkpoint for efficient decoding?

https://github.com/google/maxtext/blob/db31dd4b0b686bca4cd7cf940917ec372faa183a/end_to_end/tpu/gemma/7b/1_test_gemma.sh#L42

File: end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh [hunk 7]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# This file, combined with step 1 in the same directory, demonstrates:
```
Comment (A9isha) [inline end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh:3]: Perhaps mention this test is run at a nightly cadence

https://github.com/google/maxtext/blob/db31dd4b0b686bca4cd7cf940917ec372faa183a/end_to_end/tpu/gemma/7b/2_test_gemma.sh#L3

File: end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    echo "BASE_OUTPUT_PATH is not set, using BASE_OUTPUT_PATH = ${BASE_OUTPUT_PATH}"
+fi
+
+export M_ENABLE_CHECKPOINTING=true
+export M_BASE_OUTPUT_DIRECTORY=${BASE_OUTPUT_PATH}${MODEL_VARIATION}
+export M_DATASET_PATH=gs://maxtext-dataset
+export M_ASYNC_CHECKPOINTING=false
+
+# Run decoding
+python3 MaxText/decode.py MaxText/configs/base.yml load_parameters_path=${BASE_OUTPUT_PATH}${MODEL_VARIATION}/decode-ckpt-maxtext/0/items run_name=decoding per_device_batch_size=1 model_name=mixtral-8x7b tokenizer_path=gs://maxtext-external/mixtral-8x7B-v0.1-Instruct/tokenizer.mistral ici_tensor_parallelism=4 ici_fsdp_parallelism=16 max_prefill_predict_length=11 max_target_length=24 prompt="[INST] I love to [/INST]" autoregressive_decode_assert="That's great to hear! I love to learn new things" attention=dot_product
```
Comment (A9isha) [inline end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh:28]: Do we not want to have the full story of all the tests of decoding/pretraining/finetuning as in

end_to_end/tpu/gemma/7b/2_test_gemma.sh ?

Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh:28]: Thanks for mentioning this! We could have future PRs to add more tests. I think the main goal for this PR is to split the script into 2. Does this sound good?

File: end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh (hunk 7 above)
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh:3]: Sure, I will emphasize it's a nightly run.

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh (hunk 5 above)
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh:3]: Sure, will add nightly run.

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh (hunk 6 above)
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh:24]: I think this ckpt is used for both decoding and fine-tuning in the 2nd script. Do you suggest we add `generate_param_only_checkpoint.py` test for decoding purpose? If so, can I add more tests in a coming PR? I think the main goal of this PR to split the `test_mixtral.sh` into 2, and we could definitely add more tests. Please feel free to let me know your thoughts!

File: end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh (hunk 7 above)
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh]: Added `runs on daily basis`

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh (hunk 5 above)
Comment (RissyRan) [reply end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh]: Added `runs on daily basis`

File: end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh (hunk 6 above)
Comment (A9isha) [reply end_to_end/tpu/mixtral/8x7b/1_test_mixtral.sh:24]: yes `generate_param_only_checkpoint.py` helps getting unscanned checkpoint for faster decoding, whereas the `llama_or_mistral_ckpt.py` generates a scanned checkpoint where decoding definitely works albeit a bit slower.

Yes, please go ahead and add the other functionalities like the gemma-7b/llama2-7b tests in coming PRs. Thanks Ran!

**Review by A9isha — APPROVED** [review]
Thanks Ran

**Pull request merged**